lib/execution_engine2/utils/SDKMethodRunner.py

```diff
@@ -624,33 +637,42 @@ def finish_job(self, job_id, ctx, error_message=None, job_output=None):
             raise ValueError("Please provide valid job_id")
 
         self.check_permission_for_job(job_id=job_id, ctx=ctx, write=True)
-
         job = self.get_mongo_util().get_job(job_id=job_id)
         job_status = job.status
 
         if job_status not in [Status.running.value]:
             raise InvalidStatusTransitionException(
                 "Unexpected job status: {}".format(job_status)
             )
-
+        # Save a failed job
         if error_message:
             job.errormsg = error_message
             self.get_mongo_util().update_job_status(
                 job_id=job_id, status=Status.error.value
             )
+        # Save a successful job
         else:
-
             if not job_output:
                 raise ValueError("Missing job output for finished job")
 
             output = JobOutput()
             output.version = job_output.get("version")
-            output.id = job_output.get("id")
+            output.id = ObjectId(job_output.get("id"))
             output.result = job_output.get("result")
-            output.validate()
-            job.job_output = output
-
-            self.get_mongo_util().update_job_status(job_id=job_id, status=Status.finished.value)
+            try:
+                output.validate()
+                job.job_output = output
+                job.status = Status.finished.value
```

Comment:
redundant with `update_job_status` in next line.

Reply:
I don't understand

Reply:
`update_job_status` was called right after `job.status = Status.finished.value`